### ISSUE TITLE
feature request: tokio::io::AsyncSeek support for Reader

### ISSUE THREAD

**Issue by rbtcollins** [zkat/cacache-rs#57]
Use case: I have zip files being added to a cacache instance, and zip is a non-streamable format - the index of a zip is at the end, so it is somewhat inefficient to read everything, then scan the index, then reopen and re-read to extract the actual content (using `async_zip`)

**Comment by zkat** [comment]
sure, this sounds good.

Ideally, we'll also have the async-std version of this done at the same time, though.

**Comment by zkat** [comment]
Yeah. And we can have the unchecked and checked versions, in case you don't want to read the whole thing because, e.g. the file is enormous.

Ok.

I'll take a PR for this if anyone's interested, but it would need to be implemented for tokio, async-std, _and_ sync. I don't want feature drift between the backends.
